community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/InstrumentedGraphStatistics.scala

```diff
@@ -48,37 +48,15 @@ case class GraphStatisticsSnapshot(map: Map[StatisticsKey, Double] = Map.empty)
     snapshot.freeze
   }
 
+  //A plan has diverged if there is a relative change in any of the
+  //statistics that is bigger than the threshold
   def diverges(snapshot: GraphStatisticsSnapshot, minThreshold: Double): Boolean = {
     assert(map.keySet == snapshot.map.keySet)
-    def vectorLength(map: Map[StatisticsKey, Double]): Double =
-      Math.sqrt(map.values.map(Math.pow(_, 2)).kahanSum)
-
-    val dotProduct: Double = (map.toSeq ++ snapshot.map.toSeq)
-      .groupBy(_._1)
-      .values
-      .map(_.map(_._2).product)
-      .kahanSum
-
-    // the dot product between _any_ vector with a "zero vector" is always 0, so
-    // if the dot product is 0 we call it diverged unless both vectors are equal
-    if (dotProduct == 0) {
-      return map != snapshot.map
-    }
-    // otherwise we compute the angular difference between the two vectors, and compare to the given threshold
-    val cosine = dotProduct / (vectorLength(map) * vectorLength(snapshot.map)) // range: [0, pi/2]
-    val divergence = Math.acos(cosine) * 2 / Math.PI // range: [0, 1]
-    divergence >= minThreshold
```

Comment:
`cosine` here is actually in the range [-1, 1]
`Math.acos` returns a value in [0, pi] (or `NAN` when cosine is larger than 1 which easily can happen here due to rounding errors)
`divergence` is thus in the range [0, 2]